src/liballoc/boxed.rs

```diff
@@ -276,6 +276,13 @@ impl<T: fmt::Debug + ?Sized> fmt::Debug for Box<T> {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
+impl<T> fmt::Pointer for Box<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Pointer::fmt(&*self, f)
```

Comment:
Yeah `Box` is pretty special, and I'm not actually sure the definition has much bearing on what you can do with the type itself. You'll probably want to pull out `&T` from `&Box<T>` and then format that.